stats-test.cpp

```diff
@@ -2,22 +2,40 @@
 
 #include "catch.hpp"
 #include "stats.h"
-
+#include <cfloat>
 #include <cmath>
 
 TEST_CASE("reports average, minimum and maximum") {
     auto computedStats = Statistics::ComputeStatistics({1.5, 8.9, 3.2, 4.5});
     float epsilon = 0.001;
-    REQUIRE(std::abs(computedStats.average - 4.525) < epsilon);
-    REQUIRE(std::abs(computedStats.max - 8.9) < epsilon);
-    REQUIRE(std::abs(computedStats.min - 1.5) < epsilon);
+    REQUIRE(std::abs(computedStats.avrg - 4.525) < epsilon);
+    REQUIRE(std::abs(computedStats.maxi - 8.9) < epsilon);
+    REQUIRE(std::abs(computedStats.mini - 1.5) < epsilon);
 }
 
-TEST_CASE("average is NaN for empty array") {
-    auto computedStats = Statistics::ComputeStatistics({});
+TEST_CASE("average, maximum and minimum is NaN for empty array") {
+      auto computedStats = Statistics::ComputeStatistics({});
     //All fields of computedStats (average, max, min) must be
     //NAN (not-a-number), as defined in math.h
+    REQUIRE(std::isnan(computedStats.avrg ) == true);
+    REQUIRE(std::isnan(computedStats.maxi ) == true);
+    REQUIRE(std::isnan(computedStats.mini ) == true);
     
     //Design the REQUIRE statement here.
     //Use http://www.cplusplus.com/reference/cmath/isnan/
 }
+
+TEST_CASE("average, minimum and maximum when NaN is input") {
+    auto computedStats = Statistics::ComputeStatistics({nan(""),1.5, 8.9, 3.2, 4.5});
+    float epsilon = 0.001;
+    REQUIRE(std::abs(computedStats.avrg - 4.525) < epsilon);
+    REQUIRE(std::abs(computedStats.maxi - 8.9) < epsilon);
+    REQUIRE(std::abs(computedStats.mini - 1.5) < epsilon);
+}
+
+TEST_CASE("average, minimum and maximum when all inputs are NaN") {
+    auto computedStats = Statistics::ComputeStatistics({nan("")});
```

Comment:
When all inputs are NaN, shouldn't the statistics also be NaN? Average of NaNs is not zero...